infra/Pos/Slotting/Ntp.hs

```diff
@@ -65,12 +70,21 @@ makeLenses ''NtpSlottingState
 ----------------------------------------------------------------------------
 
 -- | Monad transformer which implements NTP-based solution for slotting.
-type NtpSlotting = Ether.ReaderT NtpSlottingVar
+-- Flag means whether to use real NTP servers or rely on local time.
+type NtpSlotting = Ether.ReaderT (Bool, NtpSlottingVar)
```

Comment:
Can you make this `Bool` flag into a separate transformer? So you'd have `Ether.ReaderT NtpEnabled (Ether.ReaderT NtpSlottingVar m)`.

I believe this way you won't need the `local'` method you define below or two separate constraints (`RealNtpSlotting will be implied by `MonadNtpSlotting`). Don't worry about the performance overhead of nested readers, I'm almost done with flattening in my branch.